src/net_processing.cpp

```diff
@@ -2124,9 +2131,18 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
         return false;
     }
 
-    if (strCommand == NetMsgType::ADDR) {
+    if (strCommand == NetMsgType::ADDR || strCommand == NetMsgType::ADDRv2) {
+        if (strCommand == NetMsgType::ADDRv2) {
+            vRecv.SetVersion(vRecv.GetVersion() | SERIALIZE_ADDR_AS_V2);
```

Comment:
This will flip `vRecv` to always deserialize addr messages as addrv2. If we subsequently receive addr(v1) then it will fail to parse it as addrv2. It would be unusual for a peer to send addrv2 followed by addrv1, but still that's not forbidden and we should handle it.